aten/src/ATen/mps/IndexKernels.h

```diff
@@ -0,0 +1,108 @@
+                                device const uint3  * offsets       [[buffer(3)]],
+                                device const void   * inputData     [[buffer(4)]],
+                                device void         * outputData    [[buffer(5)]],
+                                uint thread_index [[thread_position_in_grid]]);
+template
+[[host_name("index_select_int32")]]
+kernel void index_select<int32_t>(device const IndexAB & indexAB       [[buffer(0)]],
+                                  device const void    * indexSizes    [[buffer(1)]],
+                                  device const void    * indexStrides  [[buffer(2)]],
+                                  device const uint3   * offsets       [[buffer(3)]],
+                                  device const void    * inputData     [[buffer(4)]],
+                                  device void          * outputData    [[buffer(5)]],
+                                  uint thread_index [[thread_position_in_grid]]);
+template
+[[host_name("index_select_int64")]]
+kernel void index_select<int64_t>(device const IndexAB & indexAB       [[buffer(0)]],
+                                  device const void    * indexSizes    [[buffer(1)]],
+                                  device const void    * indexStrides  [[buffer(2)]],
+                                  device const uint3   * offsets       [[buffer(3)]],
+                                  device const void    * inputData     [[buffer(4)]],
+                                  device void          * outputData    [[buffer(5)]],
+                                  uint thread_index [[thread_position_in_grid]]);
+template
+[[host_name("index_select_int16")]]
+kernel void index_select<int16_t>(device const IndexAB & indexAB       [[buffer(0)]],
+                                  device const void    * indexSizes    [[buffer(1)]],
+                                  device const void    * indexStrides  [[buffer(2)]],
+                                  device const uint3   * offsets       [[buffer(3)]],
+                                  device const void    * inputData     [[buffer(4)]],
+                                  device void          * outputData    [[buffer(5)]],
+                                  uint thread_index [[thread_position_in_grid]]);
+template
+[[host_name("index_select_uint8")]]
+kernel void index_select<uint8_t>(device const IndexAB & indexAB       [[buffer(0)]],
+                                  device const void    * indexSizes    [[buffer(1)]],
+                                  device const void    * indexStrides  [[buffer(2)]],
+                                  device const uint3   * offsets       [[buffer(3)]],
+                                  device const void    * inputData     [[buffer(4)]],
+                                  device void          * outputData    [[buffer(5)]],
+                                  uint thread_index [[thread_position_in_grid]]);
```

Comment:
int8 is missing here, isn't it?

Reply:
Fixed in next [PR](https://github.com/kulinseth/pytorch/pull/77/)